controller/gateway/controller.go

```diff
@@ -116,9 +116,11 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	}
 
 	log.Trace(logger, "managing cleanup for gateway resource", gateway)
-	cleanupUnderway, result, err := r.cleanup(ctx, logger, &gateway)
-	if cleanupUnderway {
+	shouldReturnEarly, result, err := r.cleanup(ctx, logger, &gateway)
+	if err != nil || result.Requeue {
```

Comment:
Is `else if` required here?
Should this be clearer?
```golang
shouldReturnEarly, result, err := r.cleanup(ctx, logger, &gateway)
	if err != nil || result.Requeue {
		return result, err
	}  
        if shouldReturnEarly {
		return ctrl.Result{}, nil
	}
```

Reply:
It's not required. If you feel that this is more readable then let's go ahead with it.

Reply:
Changed this, PTAL.